cvpysdk/backupset.py

```diff
@@ -663,6 +696,278 @@ def _update(self, backupset_name, backupset_description, default_backupset):
 
         return self._process_update_reponse(request_json)
 
+    @staticmethod
+    def _get_timestamp(formatted_time):
+        """ Returns the epoch timestamp given the input time is in format %Y-%m-%d %H:%M:%S
+        
+        Args:
+            formatted_time          (multiple)  --  String of formatted time or integer of timestamp
+            
+        Returns:
+            (int)- Timestamp
+            
+        Raises:
+            SDKException:
+                If the timestamp cannot be obtained
+        """
+
+        if str(formatted_time) == '0':
+            return 0
+
+        try:  # Try converting the value to Integer, if possible then return it.
+            timestamp = int(formatted_time)
+            return timestamp
+
+        except ValueError:  # If not convertible to integer, then get the timestamp from formatted time
+            try:
+                return int(time.mktime(datetime.datetime.strptime(formatted_time, "%Y-%m-%d %H:%M:%S").timetuple()))
+            except:
+                raise SDKException('Subclient', '106')
+
+    def _set_defaults(self, custom, defaults):
+        """ Recursively sets default values on a dictionary
+
+        Args:
+            custom                  (dict)    --  The dictionary which has to be set with defaults
+            defaults                (dict)    --  The dictionary with default values
+
+        Returns:
+            None
+
+        """
+
+        for key in defaults:
+            if key not in custom:
+                custom[key] = defaults[key]
+
+            if type(defaults[key]) == dict:
+                self._set_defaults(custom[key], defaults[key])
+
+    def _prepare_browse_options(self, options):
+        """ Prepares the options for the Browse/find operation
+
+        Args:
+            options             (dict)    --   A dictionary of browse options
+
+        Returns:
+            (dict): The browse options with all the default options set
+
+        """
+
+        self._set_defaults(options, self.default_browse_options)
+
+        return options
+
+    def _prepare_browse_json(self, options):
+        """ Prepares the JSON object for the browse request
+
+        Args:
+            options                 (dict)   --    The browse options dictionary
+
+        Returns:
+            (dict): A JSON object for the browse response
+
+        """
+
+        operation_types = {
+            'browse': 0,
+            'find': 1
+        }
+
+        options['operation'] = options['operation'].lower()
+        options['operation'] = 'find' if options['operation'] not in operation_types else options['operation']
+
+        request_json = {
+            "opType": operation_types[options['operation']],
+            "mode": {
+                "mode": 2

```

Comment:
The mode cannot be set to 2 by default, as some agents are not able to browse with mode 2.
For some agents, the browse mode is 4